modules/fundamental/src/vulnerability/endpoints/mod.rs

```diff
@@ -132,3 +132,29 @@ pub async fn delete(
         Ok(HttpResponse::NotFound().finish())
     }
 }
+
+#[utoipa::path(
+  operation_id = "analyze",
+  tag = "vulnerability",
+  request_body = AnalysisRequest,
+  responses(
+      (status = 200, description = "Analyze the provided purls to search for known vulnerabilities", body = AnalysisResponse),
+  ),
+)]
+#[post("/v2/analyze")]
+/// Analyze the provided purls for the known vulnerabilities
+pub async fn analyze(
+    service: web::Data<VulnerabilityService>,
+    db: web::Data<Database>,
+    request: web::Json<AnalysisRequest>,
+    _: Require<ReadAdvisory>,
+) -> actix_web::Result<impl Responder> {
+    let purls: Vec<&str> = request.purls.iter().map(|s| s.as_str()).collect();
+    match service.analyze_purls(purls, db.as_ref()).await {
+        Ok(details) => Ok(HttpResponse::Ok().json(details)),
+        Err(error) => {
+            Ok(HttpResponse::InternalServerError()
+                .body(format!("Error analyzing purls: {}", error)))
+        }
+    }
+}
```

Comment:
To me, this belongs in the `purl` module, not vulnerability. We already have a [similar api](https://github.com/trustification/trustify/blob/5e1075324e8821c8a636725c7da004127882cd2f/modules/fundamental/src/purl/endpoints/version.rs#L11-L36) that I don't think is used by the UX, but I would prefer you make that work for you instead of adding a completely new api here.

Reply:
As this endpoint returns vulnerabilities based on purls I considered to use this one and drop the other one. Wouldn't it make more sense?